src/chaosinventory/base/models.py

```diff
@@ -1,3 +1,198 @@
+
+    type = models.ForeignKey(
+        'DataType',
+        on_delete=models.RESTRICT
+    )
+
+    class Meta:
+        abstract = True
+
+class Tag(models.Model):
+    name = models.CharField(max_length=100)
+    parent = models.ForeignKey(
+        'self',
+        on_delete = models.RESTRICT,
+        null = True,
+        blank = True,
+    )
+
+class DataType(CommonModel):
+    pass
+
+class EntityData(CommonTypeData):
+    entity = models.ForeignKey(
+        'Entity',
+        on_delete=models.CASCADE,
+    )
+
+class LocationData(CommonTypeData):
+    location = models.ForeignKey(
+        'Location',
+        on_delete=models.CASCADE,
+    )
+
+class ProductData(CommonTypeData):
+    product = models.ForeignKey(
+        'Product',
+        on_delete=models.CASCADE,
+    )
+
+class ItemData(CommonTypeData):
+    item = models.ForeignKey(
+        'Item',
+        on_delete=models.CASCADE,
+    )
+
+class Entity(CommonModel):
+    part_of = models.ForeignKey(
+        'self',
+        on_delete=models.RESTRICT,
+        null = True,
+        blank = True,
+    )
+
+    tags = models.ManyToManyField(
+        'Tag',
+    )
+
+class Location(CommonModel):
+    in_location = models.ForeignKey(
+        'self',
+        on_delete=models.RESTRICT,
+        null = True,
+        blank = True,
+    )
+
+    belongs_to = models.ForeignKey(
+        'Entity',
+        on_delete=models.RESTRICT,
+        null = True,
+        blank = True,
+    )
+
+    tags = models.ManyToManyField(
+        'Tag',
+    )
+
+class Product(CommonModel):
+    tags = models.ManyToManyField(
+        'Tag'
+    )
+
+    inventory_id = models.ForeignKey(
+        'InventoryId',
+        on_delete=models.RESTRICT,
+    )
+
+
+class Item(CommonModel):
+    amount = models.PositiveIntegerField()
+
+    belongs_to = models.ForeignKey(
+        'Entity',
+        on_delete=models.RESTRICT,
+        null = True,
+        blank = True,
+    )
+
+    in_location = models.ForeignKey(
+        'Location',
+        on_delete=models.RESTRICT,
+        related_name='contents',
+        null = True,
+        blank = True,
+    )
+
+    current_location = models.ForeignKey(
+        'Location',
+        on_delete=models.RESTRICT,
+        related_name='current_contents',
+        null = True,
+        blank = True,
+    )
+
+    product = models.ForeignKey(
+        'Product',
+        on_delete=models.RESTRICT
+    )
+
+    in_item = models.ForeignKey(
+        'Item',
+        on_delete=models.RESTRICT,
+        related_name='contents',
+        null = True,
+        blank = True,
+    )
+
+    current_item = models.ForeignKey(
+        'Item',
+        on_delete=models.RESTRICT,
+        related_name='current_contents',
+        null = True,
+        blank = True,
+    )
+
+    tags = models.ManyToManyField(
+        'Tag'
+    )
+
+    inventory_id = models.ForeignKey(
+        'InventoryId',
+        on_delete=models.RESTRICT,
+    )
+
+class InventoryIdSchema(CommonModel):
```

Comment:
How do we identify different InventoryIdSchemas?

I would propose another field `name`, referring to a plugins module name and possibly sub-classes.

E.g. `chaosinventory.base.schemes.counting`, `chaosinventory.base.schemes.letters`, ...

Reply:
Okay my bad, it inherits `name` and `note` from CommonModel...